chainerx_cc/examples/mnist_cc/mnist.cc

```diff
@@ -0,0 +1,76 @@
+#include "mnist.h"
+
+#include <cassert>
+#include <cstdint>
+#include <fstream>
+#include <memory>
+#include <string>
+
+#include "chainerx/array.h"
+#include "chainerx/device.h"
+#include "chainerx/dtype.h"
+#include "chainerx/routines/creation.h"
+#include "chainerx/shape.h"
+
+namespace chx = chainerx;
+
+namespace {
+
+chx::Array ReadArray(std::ifstream& ifs, const chx::Shape& shape) {
+    int64_t n = shape.GetTotalSize();
+
+    std::shared_ptr<uint8_t> data{new uint8_t[n], std::default_delete<uint8_t[]>{}};
+    ifs.read(reinterpret_cast<char*>(data.get()), n);
+
+    return chx::FromContiguousHostData(shape, chx::Dtype::kUInt8, static_cast<std::shared_ptr<void>>(data), chx::GetDefaultDevice());
+}
+
+int32_t ReadInt32(std::ifstream& ifs) {
+    int32_t result = 0;
+    for (int i = 0; i < 4; ++i) {
+        char byte;
+        ifs.read(&byte, sizeof(byte));
+        result = (result << 8) | (static_cast<uint32_t>(byte) & 0xff);
```

Comment:
`& 0xff` is redundant because `byte` is `char`.

```suggestion
        result = (result << 8) | static_cast<uint32_t>(byte);
```

Reply:
This mask is required.
`char` is a signed 8-bit type, so the bits 31-8 may be filled with 1.

Reply:
Oh, hmm, how about using `uint8_t` instead of `char`?
As I stated in other review, mixing signed and unsigned is confusing.

Reply:
`ifs.read` accepts only `char*` as the first parameter, so your idea requires `reinterpre_cast`.

Reply:
OK, then how about this?

```suggestion
        result = (result << 8) | static_cast<uint32_t>(static_cast<uint8_t>(byte));
```